server/jetstream_cluster.go

```diff
@@ -7296,24 +7297,49 @@ func (s *Server) jsClusteredMsgDeleteRequest(ci *ClientInfo, acc *Account, mset
 	s.sendAPIResponse(ci, acc, subject, reply, string(rmsg), s.jsonResponse(resp))
 }
 
+// saPool pools stream assignment structs for encoding.
+var saPool = sync.Pool{
+	New: func() any {
+		return &streamAssignment{}
+	},
+}
+
+func saPoolPut(sa *streamAssignment) {
+	ciPool.Put(sa.Client)
+	sa.Client = nil
```

Comment:
Do we know js lock is held here?

Reply:
The only places `saPoolPut()` are called from are places where we do `encodeAddStream()` etc, so the JS lock is held. That said, only copies of the stream assignment for encoding are returned to the pool.